WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewModel.swift

```diff
@@ -43,4 +43,34 @@ final class NotificationsViewModel {
     func didChangeDefaultAccount() {
         lastSeenTime = nil
     }
+
+    func loadNotification(
+        near note: Notification,
+        allNotifications: [Notification],
+        withIndexDelta delta: Int
+    ) -> Notification? {
```

Comment:
I'm assuming that the `viewModel` will eventually be responsible for loading and retaining the list of `allNotifications`. In that case, this method may change, to something like this:

```swift
func notification(near index: Int, delta: Int) -> Notification? {
/// `allNotifications` param was removed because it is now provided internally by the `viewModel`.
}
```

There's nothing wrong with this code, but I think it would be more beneficial to focus the refactoring efforts on areas less likely to be **re-refactored** in the future. For example, I noticed this logic is repeated in multiple places:

```swift
guard let notes = tableViewHandler.resultsController?.fetchedObjects as? [Notification] else {
    return
}
```

I think we can move it to the `viewModel`:

```swift
class NotificationsViewModel {
    private let tableViewHandler: WPTableViewHandler

    var notifications: [Notification] {
        guard let notes = tableViewHandler.resultsController?.fetchedObjects as? [Notification] else {
            return []
        }
        return notes
    }
}
```

Reply:
As we refactor the view controller to MVVM, I suggest we adopt a bottom-up approach, starting with the fundamental logic that other components will rely on. This means initially focusing on core data  operations ( not `CoreData` 😅), like fetching and handling notification. By establishing this strong foundation, we can progressively build up the architecture, ensuring each layer depends on a reliable base, and making the transition smoother and more structured.

Reply:
Yeah that's right. I am trying to keep the changes minimally intrusive while we're adding some tests and extracing pieces. Eventually VC should carry only the UI responsibilities.